### PR TITLE
SubProduct Selector Component

### PR DESCRIPTION
**Pull request by cameron-eyds**
*Issue #:*
https://github.com/bcgov/entity/issues/#16660

*Description of changes:*
* Builds generic sub product selector component
* Implementes Configuration for Mhr Sub Products
* Doesn't include integration but screenshots provided of example use.

Screenshot 2023-07-10 at 7 21 05 AM

Screenshot 2023-07-10 at 7 21 19 AM

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-auth license (Apache 2.0).

### CODE REVIEW

File: auth-web/src/components/auth/common/SubProductSelector.vue [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+      </div>
+    </v-radio-group>
+  </v-form>
+</template>
+
+<script lang="ts">
+import { Component, Emit, Prop, Vue, Watch } from 'vue-property-decorator'
+import { SubProductConfigIF } from '@/models/SubProductConfigIF'
+
+@Component({})
```
Comment (seeker25) [inline auth-web/src/components/auth/common/SubProductSelector.vue:55]: Please use composition api

File: auth-web/src/components/auth/common/SubProductSelector.vue [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+                  <ul>
+                    <li v-for="(bullet, index) in subProduct.productBullets" :key="index" class="bullet mt-2 ml-n1">
+                      <span :class="{ 'font-weight-bold': isImportantBullet(subProduct, index) }">
+                        {{ bullet }}
+                      </span>
+                    </li>
+                  </ul>
+                </p>
+                <p v-if="subProduct.note" class="sub-product-note my-6 pr-3">
+                  <strong>Note:</strong> <span v-html="subProduct.note"></span>
```
Comment (seeker25) [inline auth-web/src/components/auth/common/SubProductSelector.vue:40]: v-html should be fine I believe we control subProduct.note entirely

**Review by seeker25 — APPROVED** [review]

File: auth-web/src/components/auth/common/SubProductSelector.vue (hunk 1 above)
Comment (cameron-eyds) [reply auth-web/src/components/auth/common/SubProductSelector.vue:55]: Oh for sure, didn't realize Auth-Web was there. 
Will fix 👍

File: auth-web/src/components/auth/common/SubProductSelector.vue (hunk 2 above)
Comment (cameron-eyds) [reply auth-web/src/components/auth/common/SubProductSelector.vue:40]: You bet, it's entirely local configuration 👍

**Comment by seeker25** [comment]
The only thing I'm worried about Cam is BC Assessment or any of the other partners request subroles like this :(

**Comment by cameron-eyds** [comment]
> The only thing I'm worried about Cam is BC Assessment or any of the other partners request subroles like this :(

Totally fair!
I think as per our discussions this was about as far as the UI work would go or was there a further update with regards to this?

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2382-399f32fq.web.app

File: auth-web/src/components/auth/common/SubProductSelector.vue (hunk 2 above)
Comment (cameron-eyds) [reply auth-web/src/components/auth/common/SubProductSelector.vue:40]: @seeker25 Do you think we should sanitize here? 
Does Auth Web have a package or tool to do this already perhaps?

File: auth-web/src/components/auth/common/SubProductSelector.vue (hunk 1 above)
Comment (cameron-eyds) [reply auth-web/src/components/auth/common/SubProductSelector.vue]: Updated

File: auth-web/src/components/auth/common/SubProductSelector.vue (hunk 2 above)
Comment (seeker25) [reply auth-web/src/components/auth/common/SubProductSelector.vue:40]: No need, we have control over subProduct.note - I don't think I added it to auth-web

**Comment by pwei1018** [comment]
Temporary Url for review: https://bcregistry-account-dev--pr-2382-399f32fq.web.app

**Comment by seeker25** [comment]
Is this feature flagged somehow? Probably should have asked that before approving, because I don't think there is any backend done for this

**Comment by cameron-eyds** [comment]
> Is this feature flagged somehow? Probably should have asked that before approving, because I don't think there is any backend done for this

Good question! 
Chose not too in this instance, for the reason you just stated. No point even turning it on yet. 
We have a ticket for integration which would include connecting the dots between this component and whatever request we would need to make in order to kick off into Assets etc..

TLDR: No feature flag, not integrated into the Panels yet, the pic was just an example of such integration :) Just a stand alone component to use in the future.
